cartofacade/carto_facade_test.go

```diff
@@ -129,3 +133,404 @@ func TestRequest(t *testing.T) {
+			return errors.New("test error 2")
+		}
+		cartoFacade.carto = &carto
+
+		// returns error
+		err = cartoFacade.Stop(cancelCtx, 5*time.Second)
+		test.That(t, err, test.ShouldBeError)
+		test.That(t, err, test.ShouldResemble, errors.New("test error 2"))
+
+		carto.StopFunc = func() error {
+			time.Sleep(1 * time.Second)
+			return nil
+		}
+		cartoFacade.carto = &carto
+
+		// times out
+		err = cartoFacade.Stop(cancelCtx, 1*time.Millisecond)
+		test.That(t, err, test.ShouldBeError)
+		test.That(t, err, test.ShouldResemble, errors.New("timeout has occurred while trying to read request from cartofacade"))
+	})
+
+	cancelFunc()
+	activeBackgroundWorkers.Wait()
+}
+
+func TestTerminate(t *testing.T) {
+	lib := CartoLibMock{}
+
+	cancelCtx, cancelFunc := context.WithCancel(context.Background())
+	activeBackgroundWorkers := sync.WaitGroup{}
+
+	cfg, dir, err := GetTestConfig("mysensor")
+	algoCfg := GetTestAlgoConfig()
+	test.That(t, err, test.ShouldBeNil)
+	defer os.RemoveAll(dir)
+
+	cartoFacade := New(&lib, cfg, algoCfg)
+	carto := CartoMock{}
+	carto.TerminateFunc = func() error {
+		return nil
+	}
+	cartoFacade.carto = &carto
+	cartoFacade.Initialize(cancelCtx, 5*time.Second, &activeBackgroundWorkers)
+	defer cartoFacade.Terminate(cancelCtx, 5*time.Second)
+
+	t.Run("testing Terminate", func(t *testing.T) {
+		// success case
+		err = cartoFacade.Terminate(cancelCtx, 5*time.Second)
+		test.That(t, err, test.ShouldBeNil)
+
+		carto.TerminateFunc = func() error {
+			return errors.New("test error 3")
+		}
+		cartoFacade.carto = &carto
+
+		// returns error
+		err = cartoFacade.Terminate(cancelCtx, 5*time.Second)
+		test.That(t, err, test.ShouldBeError)
+		test.That(t, err, test.ShouldResemble, errors.New("test error 3"))
+
+		carto.TerminateFunc = func() error {
+			time.Sleep(1 * time.Second)
+			return nil
+		}
+		cartoFacade.carto = &carto
+
+		// times out
+		err = cartoFacade.Terminate(cancelCtx, 1*time.Millisecond)
+		test.That(t, err, test.ShouldBeError)
+		test.That(t, err, test.ShouldResemble, errors.New("timeout has occurred while trying to read request from cartofacade"))
+	})
+
+	cancelFunc()
+	activeBackgroundWorkers.Wait()
+}
+
+func TestAddSensorReading(t *testing.T) {
+	lib := CartoLibMock{}
+
+	cancelCtx, cancelFunc := context.WithCancel(context.Background())
+	activeBackgroundWorkers := sync.WaitGroup{}
+
+	cfg, dir, err := GetTestConfig("mysensor")
+	algoCfg := GetTestAlgoConfig()
+	test.That(t, err, test.ShouldBeNil)
+	defer os.RemoveAll(dir)
+
+	cartoFacade := New(&lib, cfg, algoCfg)
+	carto := CartoMock{}
+	carto.AddSensorReadingFunc = func(name string, reading []byte, time time.Time) error {
+		return nil
+	}
+	cartoFacade.carto = &carto
+	cartoFacade.Initialize(cancelCtx, 5*time.Second, &activeBackgroundWorkers)
+	defer cartoFacade.Terminate(cancelCtx, 5*time.Second)
+
+	t.Run("testing AddSensorReading", func(t *testing.T) {
+		timestamp := time.Date(2021, 8, 15, 14, 30, 45, 100, time.Local)
```

Comment:
lets do this in UTC time rather than local time

Reply:
udpated, should we change it [here](https://github.com/viamrobotics/viam-cartographer/blob/e3977eb274ed9595aa68c6a7f4e0c1999f1fee71/cartofacade/capi_test.go#L65) and [here](https://github.com/viamrobotics/viam-cartographer/blob/e3977eb274ed9595aa68c6a7f4e0c1999f1fee71/cartofacade/capi_test.go#L119) or does it need to be local for sensor reading

Reply:
yes please!